semantic-core/src/Data/Loc.hs

```diff
@@ -10,23 +9,17 @@ import Data.Text (Text, pack)
 import GHC.Stack
 import Source.Span
 
-data Loc = Loc
-  { locPath :: !Text
-  , locSpan :: {-# UNPACK #-} !Span
-  }
+newtype Path = Path { getPath :: Text }
```

Comment:
This should probably be moved to `semantic-source` at some point, along with e.g. `Blob`. We might also want to use the type-safe path representation at some point; cc @patrickt.

Reply:
Roger that. This would presumably be a `Path.AbsRelFile`.